jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/RFC6265SetCookieParser.java

```diff
@@ -0,0 +1,186 @@
+                    }
+                    else if (ch == ';')
+                    {
+                        name = setCookieValue.substring(offset, i).trim();
+                        if (!setAttribute(cookie, name, ""))
+                            return null;
+                        offset = i + 1;
+                        // Stay in the ATTRIBUTE_NAME state.
+                    }
+                }
+                case ATTRIBUTE_VALUE_START ->
+                {
+                    if (Character.isWhitespace(ch))
+                        continue;
+                    if (ch == '"')
+                        quoted = true;
+                    else
+                        --i;
+                    offset = i + 1;
+                    state = State.ATTRIBUTE_VALUE;
+                }
+                case ATTRIBUTE_VALUE ->
+                {
+                    if (quoted && ch == '"')
+                    {
+                        quoted = false;
+                        String value = setCookieValue.substring(offset, i).trim();
+                        if (!setAttribute(cookie, name, value))
+                            return null;
+                        offset = i + 1;
+                        state = State.ATTRIBUTE;
+                    }
+                    else
+                    {
+                        if (ch == ';')
+                        {
+                            String value = setCookieValue.substring(offset, i).trim();
+                            if (!setAttribute(cookie, name, value))
+                                return null;
+                            offset = i + 1;
+                            state = State.ATTRIBUTE_NAME;
+                        }
+                    }
+                }
+                default -> throw new IllegalStateException("invalid state " + state);
+            }
+        }
+
+        return switch (state)
+        {
+            case NAME -> null;
+            case VALUE_START -> HttpCookie.from(name, "");
+            case VALUE -> HttpCookie.from(name, setCookieValue.substring(offset, length).trim());
+            case ATTRIBUTE -> cookie.build();
+            case ATTRIBUTE_NAME -> setAttribute(cookie, setCookieValue.substring(offset, length).trim(), "") ? cookie.build() : null;
+            case ATTRIBUTE_VALUE_START -> setAttribute(cookie, name, "") ? cookie.build() : null;
+            case ATTRIBUTE_VALUE -> setAttribute(cookie, name, setCookieValue.substring(offset, length).trim()) ? cookie.build() : null;
+        };
+    }
+
+    private boolean setAttribute(HttpCookie.Builder cookie, String name, String value)
+    {
+        try
+        {
+            cookie.attribute(name, value);
+            return true;
+        }
+        catch (Throwable x)
+        {
+            return false;
+        }
+    }
+
+    private enum State
+    {
+        NAME, VALUE_START, VALUE, ATTRIBUTE, ATTRIBUTE_NAME, ATTRIBUTE_VALUE_START, ATTRIBUTE_VALUE
+    }
```

Comment:
move to start of the class as we have elsewhere

Reply:
Uh? We have private inner classes/enums always at the end of the class, as per Java recommendations and because it's an implementation details so it's not in the way.